v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java

```diff
@@ -92,6 +94,34 @@ public static JdbcIoWrapper of(JdbcIOWrapperConfig config) throws SuitableIndexN
     return new JdbcIoWrapper(tableReaders, sourceSchema);
   }
 
+  /**
+   * Set's the login timeout for the DataSource used for schema and index discoveries. This helps in
+   * early error reporting to the customer in case of unreachable or unavailable source database.
+   * The default login timeout for the {@link BasicDataSource} is infinite. Unfortunately, {@link
+   * BasicDataSource} does not directly support {@link DataSource#setLoginTimeout(int)}. This can be
+   * achieved by setting {@link BasicDataSource#setMaxWaitMillis} and connect timeout at the driver
+   * layer.
+   *
+   * @param dataSource
+   * @param config
+   */
+  @VisibleForTesting
+  protected static void setDataSourceLoginTimeout(
```

Comment:
For PostgreSQL the parameters seem to be different:

1. `loginTimeout`: how long to wait for establishment of database connection **in seconds** (default is `0`)
2. `connectTimeout`: timeout for socket connect operations in **seconds** (default is `10`)
3. `socketTimeout`: timeout for socket read operations in **seconds** (default is `0`)

Ref: https://jdbc.postgresql.org/documentation/use/#connection-parameters

Reply:
Thanks. I will add `loginTimeout` to the URL.

I think we can still have the socket level timeouts as all of these are restricted only to discovery and we are configuring in the order of minutes.